### DISCUSSION TITLE
How to save bitmap in android to gallery with path?

### DISCUSSION THREAD

**Discussion by heinhtoohtookyaw** [community/community#62561] [Mobile]
### Select Topic Area

Question

### Body

How to save bitmap in android to gallery with path? File also need to saved name differ.

**Comment by ShineThyuZan** [comment]
Use this function with bmp parameter,
`    fun saveQr(bmp: Bitmap) {
        scope.launch {
            bmp.let {
                val filePath =
                    Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
                val dir = File(filePath)
                if (!dir.exists()) dir.mkdirs()
                val file = File(dir, "yourFolderName" + Calendar.getInstance().timeInMillis + ".png")
                val fOut = FileOutputStream(file)

                bmp.compress(Bitmap.CompressFormat.PNG, 85, fOut)
                fOut.flush()
                fOut.close()
            }
        }
    }`

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
